### PR TITLE
android client extensions

### PR DESCRIPTION
**Pull request by loosebazooka**
Special handling when interacting with the android
plugin, which is NOT a standard java project.

### CODE REVIEW

**Comment by loosebazooka** [comment]
@patflynn @etanshaul anyone interested in taking a look?

File: scripts/publishing.gradle [hunk 1]
```diff
@@ -2,15 +2,15 @@ apply plugin: 'com.bmuschko.nexus'
 
 modifyPom {
   project {
-    name 'Google Cloud Platform: App Gradle Plugin'
-    description 'Gradle plugin that provides tasks for uploading, running and managing Google App Engine projects.'
-    url 'https://github.com/GoogleCloudPlatform/app-gradle-plugin'
+    name 'Endpoints Framework Gradle Plugin'
+    description 'This Gradle plugin provides tasks and configurations to build Google Cloud Endpoints Framework projects'
```
Comment (patflynn) [inline scripts/publishing.gradle]: to.. generate client endpoints framework client code and discovery docs?

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java [hunk 2]
```diff
@@ -33,14 +34,15 @@
  * Plugin definition for Endpoints Servers (on App Engine) for generation of
  * client libraries and discovery docs.
  *
- * Also provides the artifact "discovery-docs" that is a zip of all the discovery
- * docs that this server exposes (as defined in web.xml)
+ * Also provides the artifact "{@value ARTIFACT_CONFIGURATION}" that is a zip
```
Comment (patflynn) [inline src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java]: shouldn't this be {@code ..}?

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java [hunk 3]
```diff
@@ -58,19 +60,20 @@ public void apply(Project project) {
...
-        .create(ENDPOINTS_SERVER_EXTENSION, EndpointsServerExtension.class, project);
+        .create(SERVER_EXTENSION, EndpointsServerExtension.class, project);
   }
 
   private void createDiscoverDocConfiguration() {
     project.afterEvaluate(new Action<Project>() {
       @Override
       public void execute(Project project) {
-        project.getConfigurations().create("discovery-docs");
+        Configuration c = project.getConfigurations().create(ARTIFACT_CONFIGURATION);
```
Comment (patflynn) [inline src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java]: c? are we using this?

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java (hunk 2 above)
Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java]: Well, no, I want the actual value of that String to show.

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java (hunk 3 above)
Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java]: yeah I guess not. fixed

File: scripts/publishing.gradle (hunk 1 above)
Comment (loosebazooka) [reply scripts/publishing.gradle]: sure

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java (hunk 2 above)
Comment (patflynn) [reply src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java]: ah, in that case I thought it had to be {@value #ARTIFACT_CONFIGURATION}. But if that works.

**Review by patflynn — APPROVED** [review]

File: src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java (hunk 2 above)
Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/gradle/endpoints/framework/server/EndpointsServerPlugin.java]: yeah that might be the correct way to do it, intellij seemed okay with no #, I updated it anyway

**Pull request merged**